src/Psalm/Internal/Analyzer/Statements/Expression/CastAnalyzer.php

```diff
@@ -59,37 +58,33 @@ public static function analyze(
                     if (!$maybe_type->from_calculation) {
                         self::handleRedundantCast($maybe_type, $statements_analyzer, $stmt);
                     }
+
+                    if ($maybe_type->isSingleStringLiteral()) {
```

Comment:
That contradicts a previous check `$maybe_type->isInt()` above. This should be extracted from this condition

Reply:
this needs to be fixed. Your maybe_type is either an int or a string literal, not both

Reply:
Not sure if we are seeing the same version. In my last commit I changed it to an `elseif`.

Reply:
yep, you're right, clicked on the wrong link